src/collector/mod.rs

```diff
@@ -60,6 +61,70 @@ impl From<CollectorOptions> for WebSocketListenerConfig {
 	}
 }
 
+/// This type is used to distinguish different keys in the storage
+#[derive(Clone, Copy, Debug, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) enum CollectorPrefixType {
+	/// Candidate prefixed by ParachainId
+	Candidate(Option<u32>),
+	/// Relay chain block
+	Head,
+}
+
+/// Used to query storage for multiple hash types
+#[derive(Clone, Debug, Ord, PartialOrd, Eq)]
+pub(crate) struct CollectorKey {
+	/// Prefix for a specific collector key
+	pub prefix: CollectorPrefixType,
+	/// Block or a candidate hash
+	pub hash: H256,
+}
+
+impl CollectorKey {
+	pub(crate) fn new_relay_chain_block(hash: H256) -> Self {
+		Self { prefix: CollectorPrefixType::Head, hash }
+	}
+	pub(crate) fn new_parachain_candidate(hash: H256, para_id: u32) -> Self {
+		Self { prefix: CollectorPrefixType::Candidate(Some(para_id)), hash }
+	}
+	// This is used when we don't know the parachain id
+	pub(crate) fn new_generic_hash(hash: H256) -> Self {
```

Comment:
Do we really need this ? I think we can drop the Option from `CollectorPrefixType::Candidate`

Reply:
We don't know parachain id in the dispute events... There is also a convenient method to get information about a candidate by just a hash, so this function should be implemented I think (lookup without para id I mean).